java/code/src/com/redhat/rhn/common/db/datasource/xml/Package_queries.xml

```diff
@@ -861,15 +861,13 @@ SELECT P.id, P.summary, PP.name as provider
           FROM suseServerChannelsRetractedPackagesView
          WHERE sid = :sid
     ), ptfPackages AS (
-        SELECT pp.package_id
-          FROM rhnpackagecapability pc
-                  INNER JOIN rhnpackageprovides pp ON pc.id = pp.capability_id
-         WHERE pc.name = 'ptf()'
+        SELECT id as package_id
```

Comment:
you don't need this with clause for the ptfPackages and partOfPtfPackages. the select statement of the return query is already accessing the table `rhnpackage´, see we can just filter in the where clause of the main query, and let it use any index it have.